sql/hive-thriftserver/v2.3.5/src/main/java/org/apache/hive/service/cli/ColumnValue.java

```diff
@@ -199,7 +199,7 @@ public static TColumnValue toTColumnValue(TypeDescriptor typeDescriptor, Object
     case INTERVAL_DAY_TIME_TYPE:
       return stringValue((HiveIntervalDayTime) value);
     case DECIMAL_TYPE:
-      return stringValue((HiveDecimal)value, typeDescriptor);
+      stringValue(((BigDecimal)value).toPlainString());
     case BINARY_TYPE:
       String strVal = value == null ? null : UTF8String.fromBytes((byte[])value).toString();
```

Comment:
Could you remove these lines:
https://github.com/apache/spark/pull/23899/files#diff-5aed8ed671b217c7920c373fc491869aL26
https://github.com/apache/spark/pull/23899/files#diff-5aed8ed671b217c7920c373fc491869aR144-R151